extensions/org.wso2.carbon.config.configuration.maven.plugin/src/main/java/org/wso2/carbon/config/configuration/maven/plugin/ConfigDocumentMojo.java

```diff
@@ -346,11 +357,14 @@ private String createDescriptionComment(String description) {
      * @param realm class realm
      */
     private void addProjectToClasspath(ClassRealm realm) {
-        try {
-            final URL url = project.getArtifact().getFile().toURI().toURL();
-            realm.addURL(url);
-        } catch (MalformedURLException e) {
-            throw new ConfigurationMavenRuntimeException("Error when adding project to the class realm", e);
+        if (Arrays.stream(ProjectArtifactTypes.values()).anyMatch(type -> type.name().equalsIgnoreCase(project
+                .getArtifact().getType()))) {
+            try {
+                final URL url = project.getArtifact().getFile().toURI().toURL();
```

Comment:
We are only getting `project.getArtifact().getFile().toURI().toURL()` value if the dependency's maven artifact type is a jar or a bundle. For these 2 types `project.getArtifact().getFile().toURI().toURL()` always has a value.